guarddog/analyzer/analyzer.py

```diff
@@ -128,14 +122,15 @@ def analyze_metadata(self, path: str, info, rules=None, name: Optional[str] = No
         """
 
         all_rules = rules if rules is not None else self.metadata_ruleset
-        results = {}
+        results: dict[str, Optional[str]] = {}
```

Comment:
can we add a quick comment to explain what this dict will contain?

Reply:
sure, actually the way this is handled is far from ideal IMO.
I think it would be best to eventually migrate this to a @dataclass. 
Metadata and sourcecode results have different structures which adds complexity, also this value is traversing different classes and functions.

Reply:
done in 82d4381f8db9c2683cfb4b65e90c58b9d4686e83